packages/mobile/src/i18n.ts

```diff
@@ -1,36 +1,56 @@
 import locales from '@celo/mobile/locales'
 import hoistStatics from 'hoist-non-react-statics'
-import i18n from 'i18next'
+import i18n, { Resource } from 'i18next'
 import {
   initReactI18next,
   WithTranslation,
   withTranslation as withTranslationI18Next,
 } from 'react-i18next'
-import { APP_NAME, DEFAULT_APP_LANGUAGE, TOS_LINK } from 'src/config'
+import DeviceInfo from 'react-native-device-info'
+import * as RNFS from 'react-native-fs'
+import { APP_NAME, DEFAULT_APP_LANGUAGE, OTA_TRANSLATIONS_FILEPATH, TOS_LINK } from 'src/config'
 
 const TOS_LINK_DISPLAY = TOS_LINK.replace(/^https?:\/\//i, '')
 
-function getAvailableResources() {
-  const resources = {}
-  for (const [key, value] of Object.entries(locales)) {
-    Object.defineProperty(resources, key, {
-      get: () => value!.strings,
+async function getAvailableResources(
+  allowOtaTranslations: boolean,
+  otaTranslationsAppVersion: string
+) {
+  let cachedTranslations: Resource = {}
+  if (
+    allowOtaTranslations &&
+    DeviceInfo.getVersion() === otaTranslationsAppVersion &&
+    (await RNFS.exists(OTA_TRANSLATIONS_FILEPATH))
+  ) {
+    cachedTranslations = JSON.parse(await RNFS.readFile(OTA_TRANSLATIONS_FILEPATH))
+  }
+
+  const resources: Resource = {}
+  for (const [lng, value] of Object.entries(locales)) {
+    Object.defineProperty(resources, lng, {
+      get: () => ({
+        translation: cachedTranslations[lng] || value!.strings.translation,
```

Comment:
having a single root object in the cache with a single property (the language) is effectively part of the logic to store in the cache, but here the logic is kinda separate from the cache loading.  it would probably decrease the likelihood of an "oops" from a future contributor if the lng check happened above when reading from the cache (or as part of the logic loading the cache).  You might want to define a function:  `loadOtaTranslations(language: string)`

Reply:
sorry i'm not sure i understand...do you mean that something like `translation: loadOtaTranslations(lng) || value!.strings.translation` would be safer? for a filed with a singled cached language, something like `loadOtaTranslations(language: string)` would be unnecessarily reading the file multiple times to return undefined (if i understand the intention correctly), or we read the cached translations into memory and just have the `loadOtaTranslations` function handle only the `cachedTranslations[lng]` internally?

The single root object with a single language property that is written into the cache file allows us to retain the `Resource` type from i18next for the translations. This is defined to be `{ [language: string]: ResourceLanguage; }` so i'm not sure i agree that the `cachedTranslations[lng]` is very custom/unexpected?